files/bin/setup-local-disks

```diff
@@ -0,0 +1,206 @@
+      fi
+      needs_linked+=" ${unit}"
+    fi
+  done
+
+  if [[ ! -z "${prev_running}" ]]; then
+    systemctl stop ${prev_running}
+  fi
+
+  # Transfer state directories to the array, if they exist.
+  for unit in ${needs_linked}; do
+    local mount_point="/var/lib/${unit}"
+    local array_mount_point_unit="${array_mount_point}/${unit}"
+    echo "Copying ${mount_point}/ to ${array_mount_point_unit}/"
+    cp -a "${mount_point}/" "${array_mount_point_unit}/"
+    local mount_unit_name="$(systemd-escape --path --suffix=mount "${mount_point}")"
+    cat > "/etc/systemd/system/${mount_unit_name}" << EOF
+      [Unit]
+      Description=Mount ${unit} on EC2 Instance Store NVMe RAID0
+      [Mount]
+      What=${array_mount_point_unit}
+      Where=${mount_point}
+      Type=none
+      Options=bind
+      [Install]
+      WantedBy=multi-user.target
+EOF
+    systemd-analyze verify "${mount_unit_name}"
+    systemctl enable "${mount_unit_name}" --now
+  done
+
+  if [[ ! -z "${prev_running}" ]]; then
+    systemctl start ${prev_running}
+  fi
+}
+
+# Mounts and creates xfs file systems on all EC2 instance store NVMe disks
+# without existing file systems. Mounts in /mnt/k8s-disks/{1..} by default
+maybe_mount() {
+  idx=1
+  for dev in "${EPHEMERAL_DISKS[@]}"; do
+    if [[ -z "$(lsblk "${dev}" -o fstype --noheadings)" ]]; then
+      mkfs.xfs -l su=8b "${dev}"
+    fi
+    if [[ ! -z "$(lsblk "${dev}" -o MOUNTPOINT --noheadings)" ]]; then
+      echo "${dev} is already mounted."
+      continue
+    fi
+    local mount_point="${MNT_DIR}/${idx}"
+    local mount_unit_name="$(systemd-escape --path --suffix=mount "${mount_point}")"
+    mkdir -p "${mount_point}"
+    cat > "/etc/systemd/system/${mount_unit_name}" << EOF
+    [Unit]
+    Description=Mount EC2 Instance Store NVMe disk ${idx}
+    [Mount]
+    What=${dev}
+    Where=${mount_point}
+    Type=xfs
+    Options=defaults,noatime
+    [Install]
+    WantedBy=multi-user.target
+EOF
+    systemd-analyze verify "${mount_unit_name}"
+    systemctl enable "${mount_unit_name}" --now
+    idx=$((idx + 1))
+  done
+}
+
+## Main logic
+MNT_DIR="/mnt/k8s-disks"
+
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    -h | --help)
+      print_help
+      exit 0
+      ;;
+    -d | --dir)
+      MNT_DIR="$2"
+      shift
+      shift
+      ;;
+    *)                   # unknown option
+      POSITIONAL+=("$1") # save it in an array for later
+      shift              # past argument
+      ;;
+  esac
+done
+
+set +u
+set -- "${POSITIONAL[@]}" # restore positional parameters
+DISK_SETUP="$1"
+set -u
+
+if [[ "${DISK_SETUP}" != "raid0" && "${DISK_SETUP}" != "mount" ]]; then
+  echo "Valid disk setup options are: raid0 or mount"
+  exit 1
+fi
+
+disks=($(find -L /dev/disk/by-id/ -xtype l -name '*NVMe_Instance_Storage_*'))
+## Bail early if there are no ephemeral disks to setup
+if [[ "${#disks[@]}" -eq 0 ]]; then
+  echo "no ephemeral disks found, skipping disk setup"
+  exit 0
+fi
+
```

Comment:
maybe bail if this isn't being run by `root`?
```
if [ "$(id --user)" -ne 0 ]; then
  echo "Must be run as root"
  exit 1
fi
```

Reply:
good plan 👍